testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/CIBATest.java

```diff
@@ -702,6 +706,209 @@ public void testBackchannelAuthenticationFlowOfflineAccessWithoutBindingMessage(
+                attributes.put(CibaConfig.CIBA_BACKCHANNEL_CLIENT_NOTIFICATION_ENDPOINT, null);
+                attributes.put(CibaConfig.OIDC_CIBA_GRANT_ENABLED, Boolean.TRUE.toString());
+                clientRep.setAttributes(attributes);
+                clientResource.update(clientRep);
+                Assert.fail("Not expected to successfully update client");
+            } catch (BadRequestException bre) {
+                // Expected
+            }
+
+            // "Ping" mode with clientNotificationURL should success
+            attributes.put(CibaConfig.CIBA_BACKCHANNEL_CLIENT_NOTIFICATION_ENDPOINT, TestApplicationResourceUrls.cibaClientNotificationEndpointUri());
+            clientResource.update(clientRep);
+
+            clientRep = clientResource.toRepresentation();
+            attributes = Optional.ofNullable(clientRep.getAttributes()).orElse(new HashMap<>());
+            Assert.assertEquals("ping", attributes.get(CibaConfig.CIBA_BACKCHANNEL_TOKEN_DELIVERY_MODE_PER_CLIENT));
+            Assert.assertEquals(TestApplicationResourceUrls.cibaClientNotificationEndpointUri(), attributes.get(CibaConfig.CIBA_BACKCHANNEL_CLIENT_NOTIFICATION_ENDPOINT));
+
+            // Update to "Push" mode should fail
+            try {
+                attributes.put(CibaConfig.CIBA_BACKCHANNEL_TOKEN_DELIVERY_MODE_PER_CLIENT, "push");
+                clientResource.update(clientRep);
+                Assert.fail("Not expected to successfully update client");
+            } catch (BadRequestException bre) {
+                // Expected
+            }
+
+            // Update to unsupported algorithm should fail
+            try {
+                attributes.put(CibaConfig.CIBA_BACKCHANNEL_TOKEN_DELIVERY_MODE_PER_CLIENT, "ping");
+                attributes.put(CibaConfig.CIBA_BACKCHANNEL_AUTH_REQUEST_SIGNING_ALG, Algorithm.HS256);
+                clientResource.update(clientRep);
+                Assert.fail("Not expected to successfully update client");
+            } catch (BadRequestException bre) {
+                // Expected
+            }
+
+            // Should pass
+            attributes.put(CibaConfig.CIBA_BACKCHANNEL_AUTH_REQUEST_SIGNING_ALG, Algorithm.PS256);
+            clientResource.update(clientRep);
+
+            clientRep = clientResource.toRepresentation();
+            attributes = Optional.ofNullable(clientRep.getAttributes()).orElse(new HashMap<>());
+            Assert.assertEquals(Algorithm.PS256, attributes.get(CibaConfig.CIBA_BACKCHANNEL_AUTH_REQUEST_SIGNING_ALG));
+
+            // Revert algorithm
+            attributes.remove(CibaConfig.CIBA_BACKCHANNEL_AUTH_REQUEST_SIGNING_ALG);
+            clientResource.update(clientRep);
+        } finally {
+            revertCIBASettings(clientResource, clientRep);
+        }
+    }
+
+    // PING MODE TESTS
+    @Test
+    public void testPingMode_requestWithInvalidClientNotificationShouldFail() throws Exception {
```

Comment:
It seems that it does not test your modification of `SecureClientUrisExecutor`. If possible, it might be better to add it. WDYT?

Reply:
@tnorimat Good point. I've opened separate PR for it https://github.com/keycloak/keycloak/pull/8320 . Do you please have a chance to review?